mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -315,13 +315,15 @@ class Word2Vec extends Serializable with Logging {
     val syn0Global =
       Array.fill[Float](vocabSize * vectorSize)((initRandom.nextFloat() - 0.5f) / vectorSize)
     val syn1Global = new Array[Float](vocabSize * vectorSize)
+    val bcSyn0Global = sc.broadcast(syn0Global)
```

Comment:
@hhbyyh Joseph is correct. What you see only happens to work since you are running locally in one JVM.